nntrainer/layers/layer.cpp

```diff
@@ -103,7 +103,7 @@ void Layer::copy(std::shared_ptr<Layer> l) {
 sharedConstTensors Layer::forwarding_with_val(sharedConstTensors input) {
 
   for (unsigned int i = 0; i < num_inputs; ++i) {
-    net_input[i]->var = *input[i];
+    net_input[i]->var = input[i]->clone();
```

Comment:
why are we cloning here? If you want to copy the data, please copy the data, clone allocates new memory.
But I don't know we even copying the data is needed.

Reply:
> why are we cloning here? If you want to copy the data, please copy the data, clone allocates new memory.
> But I don't know we even copying the data is needed.

It needed. Cause input is updated outside function. Besides it is test case to test the internals. I guess we could give loose limitation of memory.

Reply:
works.